### ISSUE TITLE
Cannot upgrade to 2.1

### ISSUE THREAD

**Issue by brettlangdon** [Graylog2/graylog2-cookbook#68]
I am trying to upgrade Graylog2 on my servers, but unfortunately I am running into a small issue.

Previously I had setup Graylog2 with the cookbook version `"graylog2", :git => "https://github.com/Graylog2/graylog2-cookbook", :tag => "2.0"`, and now trying to switch to `"graylog2", "2.1.1"`.

I am getting an error when it is trying to install the `.deb` package:

```
dpkg: error processing archive /var/chef/cache/graylog-2.1-repository_1-1_all.deb (--install):
   trying to overwrite '/etc/apt/trusted.gpg.d/graylog-keyring.gpg', which is also in package graylog-2.0-repository 1-1
  Errors were encountered while processing:
   /var/chef/cache/graylog-2.1-repository_1-1_all.deb
```

I thought deleting `graylog-keyring.gpg` would resolve the issue, but unfortunately it has not.

I also tried manually downloading and install this `.deb`, with the same results.

I am assuming there is something silly I am missing/doing wrong here, and would love any help. I have provided some information below, more than happy to share more if needed.

### System information
```
$ uname -a
Linux ip-X-X-X-X 3.13.0-74-generic #118-Ubuntu SMP Thu Dec 17 22:52:10 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ dpkg --list | grep graylog
ii  graylog-2.0-repository               1-1                              all          Package to install Graylog 2.0 GPG key and repository
ii  graylog-server                       2.0.0-5                          all          Graylog server
```

### Chef-client output:
```
* apt_package[graylog-2.1-repository_1-1_all.deb] action install

  ================================================================================
  Error executing action `install` on resource 'apt_package[graylog-2.1-repository_1-1_all.deb]'
  ================================================================================

  Mixlib::ShellOut::ShellCommandFailed
  ------------------------------------
  Expected process to exit with [0], but received '1'
  ---- Begin output of dpkg -i --force-confold /var/chef/cache/graylog-2.1-repository_1-1_all.deb ----
  STDOUT: (Reading database ... 85675 files and directories currently installed.)
  Preparing to unpack .../graylog-2.1-repository_1-1_all.deb ...
  Unpacking graylog-2.1-repository (1-1) ...
  STDERR: dpkg: error processing archive /var/chef/cache/graylog-2.1-repository_1-1_all.deb (--install):
   trying to overwrite '/etc/apt/trusted.gpg.d/graylog-keyring.gpg', which is also in package graylog-2.0-repository 1-1
  Errors were encountered while processing:
   /var/chef/cache/graylog-2.1-repository_1-1_all.deb
  ---- End output of dpkg -i --force-confold /var/chef/cache/graylog-2.1-repository_1-1_all.deb ----
  Ran dpkg -i --force-confold /var/chef/cache/graylog-2.1-repository_1-1_all.deb returned 1

  Cookbook Trace:
  ---------------
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'

  Resource Declaration:
  ---------------------
  # In /var/chef/cache/cookbooks/graylog2/recipes/default.rb

   31: package repository_file do
   32:   action :install
   33:   source "#{Chef::Config[:file_cache_path]}/#{repository_file}"
   34:   if platform_family?('rhel')
   35:     provider Chef::Provider::Package::Rpm
   36:     options '--force'
   37:     notifies :run, 'execute[yum-clean]', :immediately
   38:   elsif platform?('ubuntu', 'debian')
   39:     provider Chef::Provider::Package::Dpkg
   40:     options '--force-confold'
   41:     notifies :run, 'execute[apt-update]', :immediately
   42:   end
   43: end

  Compiled Resource:
  ------------------
  # Declared in /var/chef/cache/cookbooks/graylog2/recipes/default.rb:31:in `from_file'

  apt_package("graylog-2.1-repository_1-1_all.deb") do
    package_name "graylog-2.1-repository_1-1_all.deb"
    provider Chef::Provider::Package::Dpkg
    action [:install]
    retries 0
    retry_delay 2
    default_guard_interpreter :default
    declared_type :package
    cookbook_name "graylog2"
    recipe_name "default"
    options "--force-confold"
    source "/var/chef/cache/graylog-2.1-repository_1-1_all.deb"
  end
```

**Comment by mariussturm** [comment]
Mh, I guess removing the repository package before the next chef run would fix it. Could you do:
`sudo apt-get remove graylog-2.0-repository` and let chef give another try?

**Comment by brettlangdon** [comment]
I did resolve this manually, by following the instructions listed here: http://docs.graylog.org/en/2.1/pages/installation/operating_system_packages.html#deb-apt

The difference I noticed was that the above instructions are using `graylog-2.1-repository_latest.deb` but the chef recipe is trying to use `graylog-2.1-repository_1-1_all.deb`. Not sure if this is the difference causing the issue or not.

**Comment by brettlangdon** [comment]
@mariussturm thanks for the suggestion, unfortunately now that I "resolved" it, I cannot test that theory. Sorry :(

**Comment by jamiebuxxx** [comment]
I also experienced the same issue while upgrading yesterday. I was able to resolve it manually as well by using the following command.

`dpkg -i --force-confold --force-overwrite /var/chef/cache/graylog-2.1-repository_1-1_all.deb`

I would like to understand whats happening here. Should the graylog-2.0 repo deletion be added to the Chef recipe is the version is upgraded?

**Comment by mariussturm** [comment]
@jamiebuxxx I think it's better to add the `--force-overwrite` option here: https://github.com/Graylog2/graylog2-cookbook/blob/master/recipes/default.rb#L40

The problem is that both repository packages provide the same files. But for apt it looks like both packages are not related to each other because of the different names. The automated deletion is not so easy, it could be that one would update from 1.x to 2.1 in that case we cant predict the name of the old package. So I would go with the forced overwrite.

**Comment by mariussturm** [comment]
Should be fixed in 2.1.2
